src/shapeFunctions.ts

```diff
@@ -20,7 +20,8 @@ export function deserialize(data: { [key: string]: any }, scale: Vertex): Shape
   else if (data.type_ == "rectangle") shape = Rect.deserialize(data, scale);
   else if (data.type_ == "roundrectangle") shape = RoundRect.deserialize(data, scale);
   else throw new Error(`${data.type_} deserialization is not implemented.`);
-  shape.deserializeStyle(data)
+  shape.deserializeStyle(data);
+  shape.referencePath = data?.reference_path ?? "#";
```

Comment:
This is the better implementation here but I feel like it is not the better one in general case. 
I think we should soon rethink the deserialize function and class methods.

Reply:
Have you something particular in mind ? I feel like there is not much to do about it

Reply:
No but i find this (my dev, not yours) not elegant.
I just put a flag in our mind here